src/NumericalAlgorithms/LinearSolver/ConvergenceCriteria.hpp

```diff
@@ -0,0 +1,110 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines DataBox tags for the linear solver
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+#include <utility>
+
+#include "NumericalAlgorithms/LinearSolver/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeString.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace LinearSolver {
+
+struct ConvergenceCriteria {
+  static constexpr OptionString help =
+      "The linear solver terminates when any of these criteria is matched.";
+
+  struct MaxIterations {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "The number of iterations exeeds this limit."};
+    static type lower_bound() noexcept { return 0; }
+    static type default_value() noexcept { return 1000; }
+  };
+
+  struct AbsoluteResidual {
+    using type = double;
+    static constexpr OptionString help = {
+        "The residual has reached this magnitude."};
+    static type lower_bound() noexcept { return 0.; }
+    static type default_value() noexcept { return 1e-14; }
+  };
+
+  struct RelativeResidual {
+    using type = double;
+    static constexpr OptionString help = {
+        "The residual has decreased by this factor."};
```

Comment:
The option descriptions in this function are confusing.

E.g. here I think you mean something like
"The solver terminates when the residual decreased by this factor."

The past tense in your text would imply, that this condition is somehow already met at the start of the iteration.

Reply:
The option strings must be very short, so I chose this style of documentation to complement the option string of the struct. I'm not convinced to change it yet 😄